habitat-lab/habitat/tasks/rearrange/rearrange_task.py

```diff
@@ -210,9 +210,14 @@ def reset(self, episode: Episode, fetch_observations: bool = True):
     @add_perf_timing_func()
     def _get_observations(self, episode):
         # Fetch the simulator observations, all visual sensors.
-        obs = self._sim._sensor_suite.get_observations(
-            self._sim.get_sensor_observations()
-        )
+        obs = self._sim.get_sensor_observations()
+
+        if not self._sim.sim_config.enable_batch_renderer:
+            # Post-process visual sensor observations
+            obs = self._sim._sensor_suite.get_observations(obs)
+        else:
+            # Keyframes are added so that the simulator state can be reconstituted later.
```

Comment:
How about a comment about how, for `enable_batch_renderer`, the equivalent post-processing will happen later in `whatever_function_name`.